staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go

```diff
@@ -103,13 +103,26 @@ func (g *Cloud) ensureInternalLoadBalancer(clusterName, clusterID string, svc *v
 	requestedIP := determineRequestedIP(svc, existingFwdRule)
 	ipToUse := requestedIP
 
-	// If the ILB already exists, continue using the subnet that it's already using.
-	// This is to support existing ILBs that were setup using the wrong subnet.
 	subnetworkURL := g.SubnetworkURL()
-	if existingFwdRule != nil && existingFwdRule.Subnetwork != "" {
-		// external LBs have an empty Subnetwork field.
-		subnetworkURL = existingFwdRule.Subnetwork
+	if g.AlphaFeatureGate.Enabled(AlphaFeatureILBCustomSubnet) {
+		// If this feature is enabled, changes to subnet annotation will be
+		// picked up and reflected in the forwarding rule.
+		// Removing the annotation will set the forwarding rule to use the default subnet.
+		if options.SubnetName != "" {
+			if existingFwdRule != nil && options.SubnetName != existingFwdRule.Subnetwork {
```

Comment:
I might be missing something -- why does the LoadBalancerIP get reused? When you change subnets, they will have different IP ranges.

Reply:
Correct, the code here only sets ipToUse to an IP specified by the user. This is to support the case where the user picks both the subnet and a specific IP. If the selected IP is not part of the subnet, the forwarding rule creation will error out.
svc.Spec.LoadBalancerIP will be empty in other cases, so we will be reseting the IP to use, if subnet changed.